worktree_commit.go

```diff
@@ -0,0 +1,243 @@
+package git
+
+import (
+	"io"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/filemode"
+	"gopkg.in/src-d/go-git.v4/plumbing/format/index"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+	"gopkg.in/src-d/go-git.v4/storage"
+	"gopkg.in/src-d/go-git.v4/utils/ioutil"
+
+	"gopkg.in/src-d/go-billy.v2"
+)
+
+// Commit stores the current contents of the index in a new commit along with
+// a log message from the user describing the changes.
+func (w *Worktree) Commit(msg string, opts *CommitOptions) (plumbing.Hash, error) {
```

Comment:
Sorry but I am complete missing. Why I will want to make to the user pass the index, the index is a plumbing concept. Not exposed on the `git` package. 

Mock and test is very easy because you can replace the storage where all is contained.

This is a small example of the jgit doing a commit with the porcelain API:
```java
Git git = new Git(db);
CommitCommand commit = git.commit();
commit.setMessage("initial commit").call();
```